### DISCUSSION TITLE
How do I assign a decimal property from Xaml?

### DISCUSSION THREAD

**Discussion by danielgruethling** [microsoft/microsoft-ui-xaml#9441] [Q&A]
I have a class Signal with the following property:
```
[ObservableProperty]
private decimal _value = 0;
```

I now want to assign a value to that property but this does not seem to work.

```
[...]
<data:Signal Value="5" />
[...]
```

I create some complex object using
```
XamlReader.Load(xamlString);
```

which contains multiple of the Signal objects.
But when the compiler tries to create a Signal object I get a:

> Failed to create a 'System.Decimal' from the text '5'. [Line: 1169 Position: 29]"

I need the property to be of type decimal and I can not assign the values after creating the objects. They have to be assigned during the XamlReader.Load([...]).
Is there some workaround I can use? I tried creating a resource with x:Decimal and using StaticResource to assign that but x:Decimal is not supported by the ancient Xaml schema WinUI3 still uses. Changing the text to "5.0" or "5,0" does not work either.

**Comment by DHancock** [comment]
An xaml file is just a text file. Even if it did support Decimals that just means that it would convert the text to a decimal for you automatically. You can do that your self in code behind:
````c#
 private Decimal _value;
 public string Value
 {
     set
     {
         _value = Convert.ToDecimal(value);
     }
 }
````
(error checking omitted)

**Reply by danielgruethling** [reply]
>You can do that your self in code behind:

That's exactly what I can not do. The whole design is created at runtime using XamlReader.Load. I can not set these in the code behind.

**Comment by danielgruethling** [comment]
I have tried a workaround from ChatGPT to create a converter and use
```
<data:Signal Value="{Binding Source='4', Converter={StaticResource StringToDecimalConverter}}" />
```
but that does not work either. Now just states: Failed to assign to property 'Namespace.Signal.Value'.

**Reply by danielgruethling** [reply]
Ok after some back and forth with ChatGPT I now have a workaround to use a second property of type string and decimal.TryParse in its set method to set the actual value. This is very dumb but it works I guess.
If there is a better workaround please let me know as I do not want this second property.
